tests/rdf-connect/ldes-client.test.ts

```diff
@@ -2989,4 +2981,927 @@ describe("Functional tests for the js:LdesClient RDF-Connect processor", () => {
+            undefined,
+            false,
+            false,
+            undefined,
+            ["http://localhost:3000/mock-ldes-linked-list-1.ttl"] // interrupt on this fragment
+        );
+
+        // Run client
+        const client1 = await exec1();
+        // Check we got some of the fragments and members
+        expect(client1.memberCount).toBe(0);
+        expect(client1.fragmentCount).toBe(2);
+
+        // Check that state was saved
+        expect(fs.existsSync("./tests/data/save.json")).toBeTruthy();
+
+        // Run a second client with the saved state
+        const outputStream2 = new SimpleStream<string>();
+        const timestamps2: number[] = [];
+
+        outputStream2.data((record) => {
+            // Check presence of SDS terms
+            expect(record.indexOf(SDS.stream)).toBeGreaterThan(0);
+            expect(record.indexOf(SDS.payload)).toBeGreaterThan(0);
+
+            // Extract timestamp property (ex:modified in this LDES)
+            const store = RdfStore.createDefault();
+            new Parser().parse(record).forEach((q) => store.addQuad(q));
+            const timestampQ = store.getQuads(null, EX.terms.modified)[0];
+            expect(timestampQ).toBeDefined();
+            // Keep track of timestamp for checking order
+            timestamps2.push(new Date(timestampQ.object.value).getTime());
+            // Check that all member data is present
+            expect(store.getQuads(null, EX.terms.subprop).length).toBeGreaterThan(0);
+        });
+
+        const exec2 = await processor(
+            outputStream2,
+            LINKED_LIST_LDES,
+            undefined,
+            undefined,
+            "ascending",
+            false,
+            undefined,
+            undefined,
+            false,
+            "./tests/data/save.json",
+            false,
+            false,
+            undefined,
+            undefined,
+            false,
+            false,
+        );
+
+        // Run client
+        const client2 = await exec2();
+        // Check that we didn't get any members but we still fetched mutable fragments
+        expect(client2.memberCount).toBe(9);
+        expect(client2.fragmentCount).toBe(3);
+        // Check result was ordered
+        const isSorted2 = timestamps2.every(
+            (v, i) => i === 0 || v >= timestamps2[i - 1],
+        );
+        expect(isSorted2).toBeTruthy();
+
+        // Clean up
+        fs.rmSync(path.resolve("./tests/data/save.json"));
+    });
+
+    test("Fetching an LDES in descending ordered, interrupting it and checking if state is saved and enforced upon resume", async () => {
+        const outputStream1 = new SimpleStream<string>();
+
+        // Hack needed because for some reason this file remains after the test
+        // when using npm test. It does not happen when using bun though ¯\_(ツ)_/¯
+        if (fs.existsSync("./tests/data/save.json")) {
+            await fs.rmSync(path.resolve("./tests/data/save.json"));
```

Comment:
Doing an await on a synchronous call is unnecessary. Also on the tests below this one.
```suggestion
           fs.rmSync(path.resolve("./tests/data/save.json"));
```

Reply:
Not relevant since https://github.com/rdf-connect/ldes-client/pull/94/commits/18b6267c1fe2cb6593cd0e46a63b3a38f21eba62